source/developers/concepts/hooks_extension_framework.rst

```diff
@@ -0,0 +1,155 @@
+|                    |                                                                        |  processing.                                                |
++--------------------+------------------------------------------------------------------------+-------------------------------------------------------------+
+| **Use cases**      |  Send an email notification when a user enrolls in a course.           |  Include additional information in an API endpoint response.|
+|                    |  an email notification.                                                |                                                             |
++--------------------+------------------------------------------------------------------------+-------------------------------------------------------------+
+
+.. _Django signals: https://docs.djangoproject.com/en/4.2/topics/signals/
+
+How to know when to use an Event or a Filter?
+----------------------------------------------
+
+When to use an Open edX Event?
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Use an Open edX Event when but not limited to:
+
+- Trigger custom logic or processing in response to specific actions within the platform, e.g., updating a search index after a course block is modified.
+- Communicate, synchronize, or coordinate with other components or services based on specific events or actions, e.g., send certificate data from LMS to credentials service to keep models up to date.
+- Integrate with external systems or services based on specific events or actions within the platform, e.g., send user data to third-party services upon registration for marketing purposes.
+
+In summary, events can be used to integrate application components with each other or with external services, allowing them to communicate, synchronize, and perform additional actions when specific triggers occur.
+
+You can review the `Open edX Events`_ documentation for more information on `how to use events`_ in your project. This documentation includes a `list of available events`_ and `how to implement event receivers`_.
+
+.. _Open edX Events: https://docs.openedx.org/projects/openedx-events/en/latest/
+.. _how to use events: https://docs.openedx.org/projects/openedx-events/en/latest/how-tos/using-events.html
+.. _list of available events: https://docs.openedx.org/projects/openedx-events/en/latest/reference/events.html
+.. _how to implement event receivers: https://docs.openedx.org/projects/openedx-events/en/latest/how-tos/using-events.html#receiving-events
+
+When to use an Open edX Filter?
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Use an Open edX Filter when but not limited to:
+
+- Enrich the data or parameters passed to a specific component, e.g., fetch reusable LTI configurations from external plugins.
+- Intercept and modify the input of a specific component, e.g., include "Edit" link to an HTML block if certain conditions are met.
```

Comment:
I'm not sure this is the best example of modifying a component. What are some others?

Reply:
All these examples were extracted from a list of real-life use cases we've compiled: https://github.com/mariajgrimaldi/openedx-events-filters-analysis/blob/main/docs/Extensions_built_using_Hooks_Extensions_Framework.pdf. Since I'm writing the use-case list, that is just a filtered version of that document, I'll get some examples from there instead.

Reply:
You can review the changes here: https://github.com/openedx/docs.openedx.org/commit/c953566eae941823a601d68f712856d501fd5119